frontend/app/controllers/Giraffe.scala

```diff
@@ -62,14 +67,27 @@ object Giraffe extends Controller {
     )
   }
 
+  def contributeRedirect = NoCacheAction { implicit request =>
+    val countryGroup = request.getFastlyCountry.getOrElse(CountryGroup.RestOfTheWorld)
+
+    val baseUrl: Uri = redirectToGiraffe(countryGroup).absoluteURL
+    val paramsToPropagate = request.queryString.map { case (k,v) => k }.toSeq
```

Comment:
Also, I think you can replace `request.queryString.map { case (k,v) => k }.toSeq` with `request.keys`